libs/common/src/auth/services/user-verification/user-verification.service.ts

```diff
@@ -64,35 +67,69 @@ export class UserVerificationService implements UserVerificationServiceAbstracti
   async verifyUser(verification: Verification): Promise<boolean> {
     this.validateInput(verification);
 
-    if (verification.type === VerificationType.OTP) {
-      const request = new VerifyOTPRequest(verification.secret);
-      try {
-        await this.userVerificationApiService.postAccountVerifyOTP(request);
-      } catch (e) {
-        throw new Error(this.i18nService.t("invalidVerificationCode"));
-      }
-    } else {
-      let masterKey = await this.cryptoService.getMasterKey();
-      if (!masterKey) {
-        masterKey = await this.cryptoService.makeMasterKey(
-          verification.secret,
-          await this.stateService.getEmail(),
-          await this.stateService.getKdfType(),
-          await this.stateService.getKdfConfig(),
-        );
-      }
-      const passwordValid = await this.cryptoService.compareAndUpdateKeyHash(
+    switch (verification.type) {
+      case VerificationType.OTP:
+        return this.verifyUserByOTP(verification);
+      case VerificationType.MasterPassword:
+        return this.verifyUserByMasterPassword(verification);
+      case VerificationType.PIN:
+        return this.verifyUserByPIN(verification);
+      case VerificationType.BIOMETRICS:
+        return this.verifyUserByBiometrics();
+    }
+  }
+
+  private async verifyUserByOTP(verification: Verification): Promise<boolean> {
+    const request = new VerifyOTPRequest(verification.secret);
+    try {
+      await this.userVerificationApiService.postAccountVerifyOTP(request);
+    } catch (e) {
+      throw new Error(this.i18nService.t("invalidVerificationCode"));
+    }
+    return true;
+  }
+
+  private async verifyUserByMasterPassword(verification: Verification): Promise<boolean> {
+    let masterKey = await this.cryptoService.getMasterKey();
+    if (!masterKey) {
+      masterKey = await this.cryptoService.makeMasterKey(
         verification.secret,
-        masterKey,
+        await this.stateService.getEmail(),
+        await this.stateService.getKdfType(),
+        await this.stateService.getKdfConfig(),
       );
-      if (!passwordValid) {
-        throw new Error(this.i18nService.t("invalidMasterPassword"));
-      }
-      this.cryptoService.setMasterKey(masterKey);
     }
+    const passwordValid = await this.cryptoService.compareAndUpdateKeyHash(
+      verification.secret,
+      masterKey,
+    );
+    if (!passwordValid) {
+      throw new Error(this.i18nService.t("invalidMasterPassword"));
+    }
+    this.cryptoService.setMasterKey(masterKey);
     return true;
   }
 
+  private async verifyUserByPIN(verification: Verification): Promise<boolean> {
+    const userKey = await this.pinCryptoService.decryptUserKeyWithPin(verification.secret);
+
+    if (userKey) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
suggestion: simplify and make intention more explicit

```suggestion
return userKey != null;
```